src/transaction/log_replication_atomic.cpp

```diff
@@ -402,37 +402,72 @@ namespace cublog
 	assert_release (false);
       }
 
-#if !defined (NDEBUG)
-    /* there will be no duplicated object in the map */
-    auto [begin, end] = m_locked_classes.equal_range (trid);
-    if (std::count_if (begin, end, [classoid] (const auto &it)
-    {
-      return it.second == *classoid;
-    }) > 0)
-    {
-      assert (false);
-    }
-#endif
+    m_locked_classes[trid].emplace (*classoid);
+  }
+
+  void
+  atomic_replicator::bookkeep_classname (cubthread::entry &thread_entry, const OID *classoid)
+  {
+    assert (!OID_ISNULL (classoid) && !OID_ISTEMP (classoid));
+    char *classname = NULL;
+
+    (void) heap_get_class_name (&thread_entry, classoid, &classname);
+    if (classname != NULL)
+      {
+	m_classname_map.emplace (*classoid, classname);
```

Comment:
I guess it needs to guarantee that a entry with classoid key in the map doesn't exist. How about adding checking logic in debug build?

Reply:
Since multiple LOG_SCHEMA_MODIFICATION_LOCKs can be replicated, an entry with classOID key can exist in the map. 

I think it is OK because `emplace` won't be performed if there is already a value for the same key (classoid)